app/attributes/attributedata.cpp

```diff
@@ -16,25 +16,26 @@
 
 #include "attributedata.h"
 
-FormItem::FormItem(
-  const QUuid &id,
-  const QgsField &field,
-  const QString &groupName,
-  const int parentTabId,
-  FormItem::FormItemType type,
-  const QString &name,
-  bool isEditable,
-  const QgsEditorWidgetSetup &editorWidgetSetup,
-  int fieldIndex,
-  const QgsExpression &visibilityExpression,
-  const QgsRelation &relation
-)
+FormItem::FormItem( const QUuid &id,
```

Comment:
It would be nice for the future to have inheritance here instead of different constructors. Like `BaseFormItem` with common fields inherited by `RelationFormItem`, `FieldFormItem`, `SpacerFormItem`,...

Reply:
Would be nice for future, agree